### PR TITLE
Reduce number of files in `mode = "local"`

### PR DESCRIPTION
**Pull request by rvhonorato**
You are about to submit a new Pull Request. Before continuing make sure you read the [contributing guidelines](CONTRIBUTING.md) and that you comply with the following criteria:

- [x] You have sticked to Python. Please talk to us before adding other programming languages to HADDOCK3
- [x] Your PR is about CNS
- [x] Your code is well documented: proper docstrings and explanatory comments for those tricky parts
- [x] You structured the code into small functions as much as possible. You can use classes if there is a (state) purpose
- [x] Your code follows our coding style
- [x] You wrote tests for the new code
- [x] `tox` tests pass. *Run `tox` command inside the repository folder*
- [ ] `-test.cfg` examples execute without errors. *Inside `examples/` run `python run_tests.py -b`*
- [x] PR does not add any dependencies, unless permission granted by the HADDOCK team
- [x] PR does not break licensing
- [x] Your PR is about writing documentation for already existing code :fire:
- [x] Your PR is about writing tests for already existing code :godmode:

---

This PR partially addresses #917 by adding a new general parameter called `less_io` - which when triggered will *not* save the `.inp` and `.out` files generated as input/output for CNS.

To achieve this I had to do some refactoring of the `libsubprocess.CNSJob.run` method to allow for polymorphism. Tried to keep it simple, if `input_file` and `output_file` are `Path` - it will follow the default old using the `TextWrapperIO` objects as input in the subprocess call.

Also had to change a bit `libparallel` - tried to simplify it for readabality and also addedd a mechanism to return the result of the `run` method of the tasks. This should make it easier for other modules using this library to not rely on filesystem operations - but I'll go deeper into that in another PR.

I've also added unittests for both `libsubprocess` and `libparallel` since there were none. Besides that I've also added an integration test for the `CNSJob`. Executing its `run` method both via file-based and memory based scheduler, to make sure it all works fine. For that I had to add a  `.pdb` and a `.psf` file, hence why the large number of lines - you can safely ignore those and filter by `.py` when reviewing

### CODE REVIEW

File: src/haddock/libs/libparallel.py [hunk 1]
```diff
@@ -183,3 +187,7 @@ def terminate(self) -> None:
             worker.terminate()
 
         log.info("The workers terminated in a controlled way")
+
+    def get_results(self) -> list:
+        """Return the results of the tasks."""
+        return self.queue.get()
```
Comment (mgiulini) [inline src/haddock/libs/libparallel.py:193]: is this necessary? I don't see it being called anywhere

File: tests/test_libparallel.py [hunk 2]
```diff
@@ -0,0 +1,120 @@
...
+            Task(1),
+            Task(2),
+            Task(3),
+        ],
+    )
+
+
+@pytest.fixture
+def scheduler_files():
+
```
Comment (mgiulini) [inline tests/test_libparallel.py:47]: can you add a couple of words explaining what this test is doing?

File: tests/test_libsubprocess.py [hunk 3]
```diff
@@ -0,0 +1,133 @@
...
+
+        yield CNSJob(
+            input_file=Path("input"),
+            output_file=Path("output"),
+            cns_exec=Path(f.name),
+        )
+
+
+def test_basejob_run(basejob, mocker):
+    basejob.make_cmd = lambda: None
```
Comment (mgiulini) [inline tests/test_libsubprocess.py:48]: same as above

**Review by mgiulini — APPROVED** [review]
very good job, clear and useful! I would just add a couple of words of description to the tricky tests for future reference (see comments)

File: tests/test_libparallel.py (hunk 2 above)
Comment (rvhonorato) [reply tests/test_libparallel.py:47]: This is the fixture, not the test - but I'll add some lines

**Review by mgiulini — APPROVED** [review]

**Review by VGPReys — APPROVED** [review]

**Pull request merged**